kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternDetailsController.java

```diff
@@ -749,40 +842,226 @@ private void addPaneToTray(Pane contentViewPane, Pane slideoutTrayPane) {
+        idx.set(0);
+        fieldDefinitions.stream().forEach(f ->
+                patternFieldList.add(idx.getAndIncrement(), new PatternField(f.meaning().description(), f.dataType(), f.purpose(), f.meaning(), "")));
+        return patternFieldList;
+    }
+
+    public ConceptEntity getLanguageConcept(PatternEntityVersion patternEntityVersion, ViewCalculator viewCalculator, SemanticEntityVersion semanticEntityVersion) {
+        int indexLang = patternEntityVersion.indexForMeaning(LANGUAGE_CONCEPT_NID_FOR_DESCRIPTION);
+
+        Object langConcept = (indexLang != -1)
+                ? semanticEntityVersion.fieldValues().get(indexLang)
+                : ENGLISH_LANGUAGE;
+
+        String langText = viewCalculator.getRegularDescriptionText(((ConceptFacade) langConcept).nid())
+                .orElse(String.valueOf(((ConceptFacade) langConcept).nid())).toUpperCase();
+
+        Set<ConceptEntity> possibleLanguages = fetchDescendentsOfConcept(getViewProperties(), LANGUAGE.publicId());
+
+        Optional<ConceptEntity> langEntity = possibleLanguages.stream().filter(lang -> lang.toString().equalsIgnoreCase(langText)).findFirst();
+
+        return langEntity.orElse(Entity.getFast(ENGLISH_LANGUAGE));
+    }
+
+    public ConceptEntity getCaseConcept(PatternEntity patternEntity, ViewCalculator viewCalculator, SemanticEntityVersion semanticEntityVersion) {
+        PatternEntityVersion patternEntityVersion = (PatternEntityVersion) viewCalculator.latest(patternEntity).get();
+        int indexCaseSig = patternEntityVersion.indexForMeaning(DESCRIPTION_CASE_SIGNIFICANCE);
+
+
+        Object caseSigConcept = (indexCaseSig != -1)
+                ? (Concept) semanticEntityVersion.fieldValues().get(indexCaseSig)
+                : NOT_APPLICABLE;
+
+        String casSigText = viewCalculator.getRegularDescriptionText(((ConceptFacade) caseSigConcept).nid())
+                .orElse(String.valueOf(((ConceptFacade) caseSigConcept).nid())).toUpperCase();
+
+        //FIXME: need a better way to query the Description semantic
+        return switch (casSigText) {
+            case "DESCRIPTION_NOT_CASE_SENSITIVE":
+                yield Entity.getFast(DESCRIPTION_NOT_CASE_SENSITIVE.nid());
+            case "DESCRIPTION_CASE_SENSITIVE":
+                yield Entity.getFast(DESCRIPTION_CASE_SENSITIVE.nid());
+            case "DESCRIPTION_INITIAL_CHARACTER_CASE_SENSITIVE":
+                yield Entity.getFast(DESCRIPTION_INITIAL_CHARACTER_CASE_SENSITIVE.nid());
+            default:
+                yield Entity.getFast(NOT_APPLICABLE.nid());
+        };
+    }
+
     public void putTitlePanesArrowOnRight() {
         putArrowOnRight(this.patternDefinitionTitledPane);
         putArrowOnRight(this.descriptionsTitledPane);
         putArrowOnRight(this.fieldsTitledPane);
     }
 
+    private Map<SemanticEntityVersion, List<String>> latestDescriptionSemantics(final ViewCalculator viewCalculator, EntityFacade conceptFacade) {
```

Comment:
Here we can try to explore the use of forEachSemanticVersionForComponentOfPattern to reduce the code footprint ?

Reply:
Agreed.
Based on our (@dholubek and I) conversation yesterday, query logic and methods on the controller should be moved into the ViewModel. The main idea of MVVM is to deal with presentation logic (no JavaFX UI objects).

Reply:
@carldea @swaroopsalvi  can we table this idea to a refactoring ticket?  this is on the critical path and we need to wrap up patterns this sprint or soon after.

Reply:
tabling for future